### PR TITLE
STORM-3090 - Fix bug when different topics use the same offset for a partition

### PR DESCRIPTION
**Pull request by choojoyq**
This pull request contains the same changes from https://github.com/apache/storm/pull/2726 that was merged in 1.x branch. As there are some conflicts during cherry-picking of these changes into 1.0.x/1.1.x branch this pull request is created for 1.0.x branch.

### CODE REVIEW

File: external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java [hunk 1]
```diff
@@ -80,22 +81,24 @@ public void refresh() {
...
-            Set<Partition> newPartitions = new HashSet<Partition>(mine);
+            Set<Partition> newPartitions = new HashSet<>(mine);
             newPartitions.removeAll(curr);
 
-            Set<Partition> deletedPartitions = new HashSet<Partition>(curr);
+            Set<Partition> deletedPartitions = new HashSet<>(curr);
             deletedPartitions.removeAll(mine);
 
-            LOG.info(taskId(_taskIndex, _totalTasks) + "Deleted partition managers: " + deletedPartitions.toString());
+            LOG.info("{}Deleted partition managers: {}",
```
Comment (srdo) [inline external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java]: Missing space

File: external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java [hunk 2]
```diff
@@ -80,22 +81,24 @@ public void refresh() {
...
+                PartitionManager manager = _managers.remove(id);
+                manager.close();
+                deletedManagers.put(new TopicAndPartition(id.topic, id.partition), manager);
             }
-            for (PartitionManager manager : deletedManagers.values()) {
-                if (manager != null) manager.close();
-            }
-            LOG.info(taskId(_taskIndex, _totalTasks) + "New partition managers: " + newPartitions.toString());
+
+            LOG.info("{}New partition managers: {}",
```
Comment (srdo) [inline external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java]: Missing space

**Comment by HeartSaVioR** [comment]
Could you please point this to at least 1.1.x-branch? We are trying to maintain 3 version lines and due to this, 1.0.x version is no longer supported.

File: external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java (hunk 2 above)
Comment (choojoyq) [reply external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java]: This space is added in ``taskId`` method.
``return "Task [" + (taskIndex + 1) + "/" + totalTasks + "] ";``

File: external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java (hunk 1 above)
Comment (choojoyq) [reply external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java]: This space is added in ``taskId`` method.
``return "Task [" + (taskIndex + 1) + "/" + totalTasks + "] ";``

**Comment by choojoyq** [comment]
base branch is changed to ``1.1.x-branch``. Am i understand correctly that the last release version of 1.0.x-branch was 1.0.6 and newer version will never be released?

File: external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java (hunk 1 above)
Comment (srdo) [reply external/storm-kafka/src/jvm/org/apache/storm/kafka/ZkCoordinator.java]: Ok, thanks.

**Comment by srdo** [comment]
+1

Yes, likely. There's no reason people can't just upgrade to 1.1.x/1.2.x instead, and it's adding a bunch of work for us to have to keep backporting to old branches.

**Comment by srdo** [comment]
Thanks @choojoyq, merged.

**Pull request merged**